platform/mbed_error.h

```diff
@@ -30,15 +30,22 @@ extern "C" {
 #endif
 
 /** Define this macro to include filenames in error context. For release builds, do not include filename to save memory.    
- *  MBED_CONF_ERROR_FILENAME_CAPTURE_ENABLED
+ *  MBED_PLATFORM_CONF_ERROR_FILENAME_CAPTURE_ENABLED
  */   
 
 /** Define this macro to disable error logging, note that the first and last error capture will still be active by default.
- *  MBED_CONF_ERROR_HIST_DISABLED  
+ *  MBED_PLATFORM_CONF_ERROR_HIST_DISABLED  
  */
     
-#ifndef MBED_CONF_MAX_ERROR_FILENAME_LEN
-#define MBED_CONF_MAX_ERROR_FILENAME_LEN            16
+#ifndef MBED_CONF_PLATFORM_MAX_ERROR_FILENAME_LEN
+#define MBED_CONF_PLATFORM_MAX_ERROR_FILENAME_LEN            16
+#else //MBED_CONF_PLATFORM_MAX_ERROR_FILENAME_LEN
+#if MBED_CONF_PLATFORM_MAX_ERROR_FILENAME_LEN > 64
+//We have to limit this to 64 bytes since we use mbed_error_printf for error reporting
+//and mbed_error_vfprintf uses 128bytes internal buffer which may not be sufficient for anything
+//longer that 64 bytes with the current implementation.
+#error "Unsupported error filename buffer length detected, max supported length is 64 chars. Please change MBED_CONF_PLATFORM_MAX_ERROR_FILENAME_LEN or max-error-filename-len in configuration."
```

Comment:
@kjbracey-arm - Please see this change to limit the filename buffer to 64 chars. This should take care of 128 chars buffer limitation in mbed_error_printf implementation.